### DISCUSSION TITLE
How does one change the local image captioning model?

### DISCUSSION THREAD

**Discussion by minipasila** [SillyTavern/SillyTavern#3109] [Q&A]
I do not want to use any API since it already has like a local option which automatically uses Xenova/vit-gpt2-image-captioning model which sucks. So I thought I'd just replace the two files /src/transformer.js and /default/config.yaml where it references that model to onnx-community/Florence-2-base-ft and somehow it still tries to use the other model even though I changed it. So is there something I'm missing as to why it keeps using the default model regardless of my changes? (This isn't the extras module.)

**Comment by minipasila** [comment]
So what I ended up doing was replacing the downloaded model files that were in \SillyTavern\data\_cache\Xenova\vit-gpt2-image-captioning\ with the Florence 2 model files and that managed to make it use that model instead finally. Though it didn't work but at least it tried to use it.

**Comment by Cohee1207** [comment]
The version of transformer.js used is too old to support Florence 2. I can't update it due to it requiring [sharp](https://www.npmjs.com/package/sharp) to function which is incompatible with Android.
